base/generator.jl

```diff
@@ -108,11 +108,50 @@ Base.HasEltype()
 iteratoreltype(x) = iteratoreltype(typeof(x))
 iteratoreltype(::Type) = HasEltype()  # HasEltype is the default
 
+abstract type IteratorAccess end
+struct ForwardAccess <: IteratorAccess end
+struct RandomAccess <: IteratorAccess end
+struct WritableRandomAccess <: IteratorAccess end
+
+"""
+    iteratoraccess(itertype::Type) -> IteratorAccess
+
+Given the type of an iterator, returns one of the following values:
+
+* `ForwardAccess()` if the iterator can be iterated over.
+* `RandomAccess()` if the iterator supports read-only indexing.
+* `WritableRandomAccess()` if the iterator supports read-write indexing.
+
+The default value (for iterators that do not define this function) is `ForwardAccess()`.
+
+```jldoctest
+julia> Base.iteratoraccess(1:5)
+Base.RandomAccess()
+
+julia> Base.iteratoraccess([1, 2, 3])
+Base.WritableRandomAccess()
+
+julia> Base.iteratoraccess(drop([1, 2, 3], 1))
+Base.WritableRandomAccess()
+```
+"""
+iteratoraccess(x) = iteratoraccess(typeof(x))
```

Comment:
You could spell `iteratoraccess` as `IteratorAccess` instead. This would be consistent with some other recent [traits](https://github.com/JuliaLang/julia/blob/master/base/traits.jl), this is spelled `IteratorAccess`. Not sure if there is an official recommendation somewhere.

Reply:
This follows `iteratorsize` etc.. I could be changed for all iterator traits at once in a different PR.

Reply:
Iteratortraits are also spelled in camel case by now, https://github.com/JuliaLang/julia/pull/25402